src/components/IntegrationsView/IntegrationDocker/DockerContainerCard.tsx

```diff
@@ -0,0 +1,248 @@
+import {
+  Badge,
+  Card,
+  Code,
+  DataList,
+  DropdownMenu,
+  Flex,
+  Heading,
+  IconButton,
+} from "@radix-ui/themes";
+import { DockerActionPayload, DockerContainer } from "../../../services/refact";
+import { useState, type FC } from "react";
+import { CopyIcon, DotsVerticalIcon } from "@radix-ui/react-icons";
+import { toPascalCase } from "../../../utils/toPascalCase";
+import { Markdown } from "../../Markdown";
+import { Link } from "../../Link";
+import { fallbackCopying } from "../../../utils/fallbackCopying";
+import { TruncateRight } from "../../Text/TruncateRight";
+import * as Collapsible from "@radix-ui/react-collapsible";
+import { Reveal } from "../../Reveal";
+import { Chevron } from "../../Collapsible";
+
+type DockerContainerCardProps = {
+  container: DockerContainer;
+  currentContainerAction: DockerActionPayload | null;
+  isActionInProgress: boolean;
+  handleDockerContainerActionClick: (
+    payload: DockerActionPayload,
+  ) => Promise<void>;
+};
+
+const DOCKER_ACTIONS: (Omit<DockerActionPayload, "container"> & {
+  label: string;
+  loadingLabel: string;
+})[] = [
+  {
+    label: "Start container",
+    loadingLabel: "Starting...",
+    action: "start",
+  },
+  {
+    label: "Stop container",
+    loadingLabel: "Stopping...",
+    action: "stop",
+  },
+  {
+    label: "Kill container",
+    loadingLabel: "Killing...",
+    action: "kill",
+  },
+  {
+    label: "Remove container",
+    loadingLabel: "Removing...",
+    action: "remove",
+  },
+];
+
+export const DockerContainerCard: FC<DockerContainerCardProps> = ({
+  container,
+  currentContainerAction,
+  isActionInProgress,
+  handleDockerContainerActionClick,
+}) => {
+  const [detailsOpen, setDetailsOpen] = useState(false);
+
+  // needed to handle disabled state of buttons accordingly to the status of docker container
+  const isDockerActionButtonDisabled = (
+    el: DockerContainer,
+    action: string,
+  ) => {
+    return (
+      (isActionInProgress && currentContainerAction?.container === el.name) ||
+      (el.status === "running" && action === "start") ||
+      (el.status === "exited" && (action === "stop" || action === "kill"))
+    );
+  };
+
+  const handleClickOnAction = ({
+    container,
+    action,
+  }: {
+    container: string;
+    action: DockerActionPayload["action"];
+  }) => {
+    void handleDockerContainerActionClick({
+      container,
+      action,
+    });
+  };
+
+  const formattedMarkdown = (text: string) => {
+    return "```" + text + "```";
```

Comment:
no newlines needed?

Reply:
actually not, since we are just displaying it inline

Reply:
should it be single back-ticks if it's inline?